tools/src/main/java/org/apache/kafka/trogdor/coordinator/Coordinator.java

```diff
@@ -235,6 +237,10 @@ public void beginShutdown() {
         lock.lock();
         try {
             this.shutdown = true;
+            for (Iterator<Fault> iter = pendingFaults.iterateByStart();
```

Comment:
To clarify, my suggestion was to include this only if we don't have the extra `shutdown` check following the `await` in the loop above. Unless there's a specific reason to have both, I'd just do one of them.

Also, it seems like you reverted the change to reset `now` after the `await`. Was that intentional?

Reply:
Sorry.  I unintentionally overwrote the earlier changes.  Fixed.